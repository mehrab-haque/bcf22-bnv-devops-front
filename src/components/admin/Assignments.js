import React from 'react'

const Assignments= props=>{
    return(
        <div>

        </div>
    )
}

export default Assignments