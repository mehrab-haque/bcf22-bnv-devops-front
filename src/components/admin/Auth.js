import React, {useRef, useState} from 'react'
import {
    Button,
    Dialog,
    DialogActions,
    DialogContent,
    DialogTitle,
    Grid,
    LinearProgress,
    TextField
} from "@mui/material";
import {showToast} from "../../App";

const Auth=props=>{

    const loginRef=useRef();
    const passRef=useRef();

    const [loading,setLoading]=useState(false)

    const loginClick=()=>{
        setLoading(true)
        setTimeout(()=>{
            setLoading(false)
            props.login()
        }, 2000);
    }


    return(
        <div>
            <Dialog open={true}>
                {
                    loading?(
                        <LinearProgress/>
                    ):(
                        <div/>
                    )
                }
                <DialogTitle>
                    Admin Login
                </DialogTitle>
                <DialogContent>
                    <Grid container spacing={1}>
                        <Grid item xs={12}>
                            <TextField
                                inputRef={loginRef}
                                fullWidth
                                variant={'outlined'}
                                label={'Login'}
                                />
                        </Grid>
                        <Grid item xs={12}>
                            <TextField
                                inputRef={passRef}
                                fullWidth
                                type={'password'}
                                variant={'outlined'}
                                label={'Password'}
                            />
                        </Grid>
                    </Grid>
                </DialogContent>
                <DialogActions>
                    <Button
                        color={'primary'}
                        onClick={loginClick}
                        disabled={loading}
                        >
                        Login
                    </Button>
                </DialogActions>
            </Dialog>
        </div>
    )
}

export default Auth