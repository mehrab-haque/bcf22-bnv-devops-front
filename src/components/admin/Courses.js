import React from 'react'

const Courses= props=>{
    return(
        <div>

        </div>
    )
}

export default Courses