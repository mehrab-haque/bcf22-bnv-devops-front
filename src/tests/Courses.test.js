import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Auth from "../components/admin/Auth";
import Assignments from "../components/admin/Assignments";
import Courses from "../components/admin/Courses";

describe("Component Courses Render", () => {
    it("renders without crashing given the required props", () => {
        const wrapper = shallow(<Courses/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
