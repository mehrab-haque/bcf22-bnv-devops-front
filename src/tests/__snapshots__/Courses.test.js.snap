// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Courses Render renders without crashing given the required props 1`] = `<div />`;
